src/os/bluestore/BlueStore.cc

```diff
@@ -2196,7 +2196,8 @@ void BlueStore::ExtentMap::reshard(
     if (spanning_blob_map.empty()) {
       bid = 0;
     } else {
-      bid = spanning_blob_map.rbegin()->first + 1;
+      int last_bid = spanning_blob_map.rbegin()->first;
```

Comment:
instead of hardwiring the type info here, i'd recommend use the type of `Blob::id` to define `bid` and check for `-1` after increment.

```diff
@@ -2208,11 +2209,13 @@ void BlueStore::ExtentMap::reshard(
     } else {
       shard_end = sp->offset;
     }
-    int bid;
+    decltype(Blob::id) bid;
     if (spanning_blob_map.empty()) {
       bid = 0;
     } else {
       bid = spanning_blob_map.rbegin()->first + 1;
+      if (bid == -1) // -1 for non-spanning blobs
+        bid = 0;
     }
     Extent dummy(needs_reshard_begin);
     for (auto e = extent_map.lower_bound(dummy); e != extent_map.end(); ++e) {
@@ -2268,6 +2271,8 @@ void BlueStore::ExtentMap::reshard(
      }
      if (must_span) {
        b->id = bid++;
+       // assuming the bid does not wrap
+       assert(b->id >= 0);
        spanning_blob_map[b->id] = b;
        dout(20) << __func__ << "    adding spanning " << *b << dendl;
      }
```